src/theory/bags/solver_state.h

```diff
@@ -41,16 +41,35 @@ class SolverState : public TheoryState
 
   /**
    * @param n has the form (bag.count e A)
-   * @pre bag A needs is already registered using registerBag(A)
+   * @pre bag A is already registered using registerBag(A)
    * @return a lemma (= skolem (bag.count eRep ARep)) where
    * eRep, ARep are representatives of e, A respectively
    */
   Node registerCountTerm(TNode n);
+
+  /**
+   * @param n has the form (bag.card A)
+   */
+  Node registerCardinalityTerm(TNode n);
+
+  /**
+   * @param n has the form (bag.card A)
+   */
+  Node getCardinalitySkolem(TNode n);
+
+  bool hasCardinalityTerms() const;
+
   /** get all bag terms that are representatives in the equality engine.
    * This function is valid after the current solver is initialized during
    * postCheck. See SolverState::initialize and BagSolver::postCheck
    */
   const std::set<Node>& getBags();
+
+  /**
+   * get all cardinality terms
```

Comment:
Can you say a bit more what is stored in the map?

Reply:
Done.